src/TSTransformer/classes/MultiTransformState.ts

```diff
@@ -13,18 +13,26 @@ export class MultiTransformState {
 	public readonly getModuleExportsAliasMapCache = new Map<ts.Symbol, Map<ts.Symbol, string>>();
 
 	public hasCheckedJsxFactory = false;
-	public checkJsxFactory(node: ts.Node, compilerOptions: ts.CompilerOptions) {
-		this.hasCheckedJsxFactory = true;
-		if (!this.hasCheckedJsxFactory && compilerOptions.jsxFactory !== "Roact.createElement") {
-			DiagnosticService.addDiagnostic(errors.invalidJsxFactory(node));
+	public isRoactJsxFactory(node: ts.Node, compilerOptions: ts.CompilerOptions) {
```

Comment:
I think I prefer the original `check-` prefix here.

An `is-` prefix indicates to me that a function checks the status of something (and probably returns a boolean) without side effects.

Reply:
Then again, check implies it does not return anything. maybe `checkIsRoactJsxFactory`?